dart/gui/osg/render/LineSegmentShapeNode.cpp

```diff
@@ -207,18 +210,16 @@ void LineSegmentShapeDrawable::refresh(bool firstTime)
     const common::aligned_vector<Eigen::Vector2i>& connections
         = mLineSegmentShape->getConnections();
 
-    ::osg::ref_ptr<::osg::DrawElementsUInt> elements
-        = new ::osg::DrawElementsUInt(::osg::PrimitiveSet::LINES);
-    elements->reserve(2 * connections.size());
+    mElements->resize(2 * connections.size());
```

Comment:
I don't think we want to use `resize(~)` alongside `push_back(~)`, because that means we're going to make `mElements` the size `2 * connections.size()` and then later push back an additional `2 * connections.size()` number of new elements.

I would recommend using `reserve(2 * connections.size()` and then `clear()`.

Reply:
Alternatively `resize(~)` and then `[2*i] = ` and `[2*i+1] = ` would also work, but that feels more risky to me.

Reply:
Thanks for catching this out. Let me fix this as suggested.